frontend/src/Components/Login/Login.jsx

```diff
@@ -1,6 +1,151 @@
-import React from "react";
+import React, { useEffect, useRef, useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
 import "./Login.css";
+import logoImage from "./assets/logoblackandwhite.svg";
+import AxiosInstance from "../axios/AxiosInstance";
 
-export const Login = () => {
-  return <div>SkyNest</div>;
+const Login = ({ setAccessToken }) => {
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [errorMsg, setErrorMsg] = useState("");
+	const [showPassword, setShowPassword] = useState(false);
+	const [loading, setLoading] = useState(false);
+	const [successfulLogin, setSuccessfulLogin] = useState("");
+
+	const emailRef = useRef();
+	const passwordRef = useRef();
+
+	const navigate = useNavigate();
+
+	const homePageRedirect = (delay) => {
+		setTimeout(() => {
+			navigate("/home");
+		}, delay);
+	};
+
+	const getUserToken = async () => {
+		await AxiosInstance.post("/login", { email, password })
+			.then(({ data }) => {
+				if (data?.accessToken) {
+					setAccessToken(data.accessToken);
+					localStorage.setItem("accessToken", data.accessToken);
+					setSuccessfulLogin("Login Successful, redirecting.");
+					homePageRedirect(2500);
+				} else {
+					setErrorMsg("Internal error, please try again");
+				}
+			})
+			.catch(({ response }) => {
+				if (response.status === 400) {
+					setErrorMsg("Incorrect username or password");
+				} else if (response.status === 0) {
+					setErrorMsg("Server Timeout");
+				}
+				setLoading(false);
+			});
+	};
+
+	useEffect(() => {
+		emailRef.current.focus();
+	}, []);
+
+	useEffect(() => {
+		setErrorMsg("");
+	}, [email, password]);
+
+	const onFormSubmit = async (e) => {
+		e.preventDefault();
+		disableEnableInputs(true);
+		setLoading(true);
+		await getUserToken();
+		disableEnableInputs(false);
+	};
+
+	const onEmailChange = (e) => {
+		setEmail(e.target.value);
+	};
+
+	const onPasswordChange = (e) => {
+		setPassword(e.target.value);
+	};
+
+	const passwordShowHide = () => {
+		setShowPassword(!showPassword);
+	};
+
+	const disableEnableInputs = (value) => {
+		emailRef.current.disabled = value;
+		passwordRef.current.disabled = value;
+	};
+
+	return (
+		<div className="container-fluid vh-100 d-flex justify-content-center align-items-center  latte">
```

Comment:
extra space before latte. What is latte class doing? Maybe more descriptive name would do it better.

Reply:
Oops, fixed.